tests/appsec/test_automated_login_events.py

```diff
@@ -1348,3 +1218,703 @@ def validate_iden(meta):
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+
+            # optional (to review for each library)
+            if context.library not in libs_without_user_exist:
+                assert meta["appsec.events.users.login.failure.usr.exists"] == "true"
+
+            if context.library not in libs_without_user_id_on_failure:
+                assert meta["appsec.events.users.login.failure.usr.id"] == USER_HASH
+                assert meta["_dd.appsec.usr.id"] == USER_HASH
+
+    def setup_login_wrong_password_failure_basic(self):
+        self.r_wrong_user_failure = weblog.get(
+            "/login?auth=basic", headers={"Authorization": BASIC_AUTH_INVALID_PASSWORD_HEADER}
+        )
+
+    @missing_feature(context.library == "php", reason="Basic auth not implemented")
+    def test_login_wrong_password_failure_basic(self):
+        assert self.r_wrong_user_failure.status_code == 401
+        for _, trace, span in interfaces.library.get_spans(request=self.r_wrong_user_failure):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.failure.usr.login"] == USERNAME_HASH
+            assert meta["_dd.appsec.usr.login"] == USERNAME_HASH
+            assert meta["_dd.appsec.events.users.login.failure.auto.mode"] == "anonymization"
+            assert meta["appsec.events.users.login.failure.track"] == "true"
+
+            # optional (to review for each library)
+            if context.library not in libs_without_user_exist:
+                assert meta["appsec.events.users.login.failure.usr.exists"] == "true"
+
+            if context.library not in libs_without_user_id_on_failure:
+                assert meta["appsec.events.users.login.failure.usr.id"] == USER_HASH
+                assert meta["_dd.appsec.usr.id"] == USER_HASH
+
+    def setup_login_sdk_success_local(self):
+        self.r_sdk_success = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser",
+            data=login_data(context, USER, PASSWORD),
+        )
+
+    def test_login_sdk_success_local(self):
+        assert self.r_sdk_success.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_sdk_success):
+            assert_priority(span, trace)
+            meta = span.get("meta", {})
+
+            # mandatory
+            assert meta["appsec.events.users.login.success.usr.login"] == "sdkUser"
+            assert meta["_dd.appsec.usr.login"] == USERNAME_HASH
+            assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "anonymization"
+            assert meta["appsec.events.users.login.success.track"] == "true"
+            assert meta["_dd.appsec.events.users.login.success.sdk"] == "true"
+
+            # optional (to review for each library)
+            if context.library not in libs_without_user_id:
+                assert meta["usr.id"] == USER_HASH
```

Comment:
just like you're checking `assert meta["appsec.events.users.login.success.usr.login"] == "sdkUser"` just above, the usr.id tag should be set by the SDK and thus not hashed.
```suggestion
                assert meta["usr.id"] == "sdkUser"
```